lib/IDE/CodeCompletion.cpp

```diff
@@ -6477,10 +6479,54 @@ static void addConditionalCompilationFlags(ASTContext &Ctx,
   }
 }
 
+static void postProcessResults(ArrayRef<CodeCompletionResult *> results,
+                               CompletionKind Kind, DeclContext *DC) {
+  auto isExprKeyword = [](const CodeCompletionResult *result) {
+    if (result->getKind() != CodeCompletionResult::ResultKind::Keyword)
+      return false;
+    switch (result->getKeywordKind()) {
+#define POUND_DIRECTIVE_KEYWORD(kw)
+#define POUND_CONFIG(kw)
+#define POUND_KEYWORD(kw) case CodeCompletionKeywordKind::pound_##kw:
```

Comment:
What about other expression-context things like `try`, `await`, `true`, `false` or statement-context like `for`, `while`, etc.

Reply:
Moved the logic to `addExprKeywords()`, `addValueLiteralCompletions()`, etc.
'ExpressionAtNonScriptOrMainFileScope' is not exactly a correct word, but I applied it to statement keyword too.